### ISSUE TITLE
second stage training with 1.5 pipeline

### ISSUE THREAD

**Issue by Rupanshparth** [bytedance/LatentSync#202]
Second stage training with latest 1.5 pipeline on a subset of voxceleb2 is predicting mask, anyone facing same issue? figured out the bug? this was after 10k steps, a while later it just predicts the mask

https://github.com/user-attachments/assets/75accf54-b3c5-459b-b946-ba265d732569

**Comment by MStypulkowski** [comment]
Yes, I'm facing the same problem. @chunyu-li do you know what can be the issue here?

**Comment by km2u-dev** [comment]
Whats the solution

**Comment by Ignorant98** [comment]
> Second stage training with latest 1.5 pipeline on a subset of voxceleb2 is predicting mask, anyone facing same issue? figured out the bug? this was after 10k steps, a while later it just predicts the mask使用最新的 1.5 管道在 voxceleb2 子集上进行的第二阶段训练正在预测口罩，有人遇到同样的问题吗？找出错误了吗？这是在 10k 步之后，过了一会儿它只是预测口罩
> 
>  val_365k_2nd.mp4

你遇到的问题可能和我一样。
#195 
#12

**Comment by chunyu-li** [comment]
It looks like that some gradients became NaN during the training process, and gradually all weights also turned into NaN.

Can you show me your stage2 training config file?

**Comment by Rupanshparth** [comment]
I used the default stage2.yaml file, resume global step was 360k from first stage ckpt

`data:
  syncnet_config_path: configs/syncnet/syncnet_16_pixel_attn.yaml
  train_output_dir: /data/s3/checkpoints
  train_fileslist: ""
  train_data_dir: "/data/s3/training_data"
  audio_embeds_cache_dir: 
  audio_mel_cache_dir: mel_new

  val_video_path: assets/demo1_video.mp4
  val_audio_path: assets/demo1_audio.wav
  batch_size: 4 # 4
  num_workers: 12 # 12
  num_frames: 16
  resolution: 256
  mask: fix_mask
  mask_image_path: latentsync/utils/mask.png
  audio_sample_rate: 16000
  video_fps: 25
  audio_feat_length: [2, 2]

ckpt:
  resume_ckpt_path: /data/s3/checkpoints/train-2025_03_24-10:25:02/checkpoints/checkpoint-360000.pt 
  save_ckpt_steps: 5000

run:
  pixel_space_supervise: true
  use_syncnet: true
  sync_loss_weight: 0.05
  perceptual_loss_weight: 0.1 # 0.1
  recon_loss_weight: 1 # 1
  guidance_scale: 1.5 # 1.5 or 1.0
  trepa_loss_weight: 10
  inference_steps: 20
  trainable_modules:
    - motion_modules.
    - attentions.
  seed: 1247
  use_mixed_noise: true
  mixed_noise_alpha: 1 # 1
  mixed_precision_training: true
  enable_gradient_checkpointing: true
  max_train_steps: 10000000
  max_train_epochs: -1

optimizer:
  lr: 1e-5
  scale_lr: false
  max_grad_norm: 1.0
  lr_scheduler: constant
  lr_warmup_steps: 0

model:
  act_fn: silu
  add_audio_layer: true
  attention_head_dim: 8
  block_out_channels: [320, 640, 1280, 1280]
  center_input_sample: false
  cross_attention_dim: 384
  down_block_types:
    [
      "CrossAttnDownBlock3D",
      "CrossAttnDownBlock3D",
      "CrossAttnDownBlock3D",
      "DownBlock3D",
    ]
  mid_block_type: UNetMidBlock3DCrossAttn
  up_block_types:
    [
      "UpBlock3D",
      "CrossAttnUpBlock3D",
      "CrossAttnUpBlock3D",
      "CrossAttnUpBlock3D",
    ]
  downsample_padding: 1
  flip_sin_to_cos: true
  freq_shift: 0
  in_channels: 13 # 49
  layers_per_block: 2
  mid_block_scale_factor: 1
  norm_eps: 1e-5
  norm_num_groups: 32
  out_channels: 4 # 16
  sample_size: 64
  resnet_time_scale_shift: default # Choose between [default, scale_shift]

  use_motion_module: true
  motion_module_resolutions: [1, 2, 4, 8]
  motion_module_mid_block: false
  motion_module_decoder_only: false
  motion_module_type: Vanilla
  motion_module_kwargs:
    num_attention_heads: 8
    num_transformer_block: 1
    attention_block_types:
      - Temporal_Self
      - Temporal_Self
    temporal_position_encoding: true
    temporal_position_encoding_max_len: 24
    temporal_attention_dim_div: 1
    zero_initialize: true
`

**Comment by chunyu-li** [comment]
You can change `use_syncnet ` to `false`, and then run 10k steps to see if the black mask still exists.

**Comment by MStypulkowski** [comment]
@Ignorant98 From what I understand from #12, SyncNet needs to be retrained only if I want to train the model on higher resolution data. In my case, I want to recreate the results on 256p videos.

**Comment by chunyu-li** [comment]
> [@Ignorant98](https://github.com/Ignorant98) From what I understand from [#12](https://github.com/bytedance/LatentSync/issues/12), SyncNet needs to be retrained only if I want to train the model on higher resolution data. In my case, I want to recreate the results on 256p videos.

I recommend first completing Stage 1 and Stage 2 training at a 256 resolution to reproduce the results in the demo before attempting to train a 512 resolution model. Since this code has not been adapted for 512 resolution training, I am also uncertain about the potential issues that may arise.

**Comment by Rupanshparth** [comment]
I ran the training with same config except I changed the resume global step to 0, now the results are improving, it could be that the optimiser's initialisation was effected by the global step

**Comment by chunyu-li** [comment]
I tried using this config file for stage2 training yesterday, and the results were good. I am unable to reproduce the black mask results you demonstrated.

```yaml
data:
  syncnet_config_path: configs/syncnet/syncnet_16_pixel_attn.yaml
  train_output_dir: debug/unet
  train_fileslist: /mnt/bn/maliva-gen-ai-v2/chunyu.li/fileslist/data_v9_syncnet.txt
  train_data_dir: ""
  audio_embeds_cache_dir: /mnt/bn/maliva-gen-ai-v2/chunyu.li/audio_cache/whisper_new
  audio_mel_cache_dir: /mnt/bn/maliva-gen-ai-v2/chunyu.li/audio_cache/mel_new

  val_video_path: assets/demo1_video.mp4
  val_audio_path: assets/demo1_audio.wav
  batch_size: 4 # 4
  num_workers: 12 # 12
  num_frames: 16
  resolution: 256
  mask: fix_mask
  mask_image_path: latentsync/utils/mask.png
  audio_sample_rate: 16000
  video_fps: 25
  audio_feat_length: [2, 2]

ckpt:
  resume_ckpt_path: checkpoints/latentsync_unet.pt
  save_ckpt_steps: 10000

run:
  pixel_space_supervise: true
  use_syncnet: true
  sync_loss_weight: 0.05
  perceptual_loss_weight: 0.1 # 0.1
  recon_loss_weight: 1 # 1
  guidance_scale: 1.5 # 1.5 or 1.0
  trepa_loss_weight: 10
  inference_steps: 20
  trainable_modules:
    - motion_modules.
    - attentions.
  seed: 1247
  use_mixed_noise: true
  mixed_noise_alpha: 1 # 1
  mixed_precision_training: true
  enable_gradient_checkpointing: true
  max_train_steps: 10000000
  max_train_epochs: -1

optimizer:
  lr: 1e-5
  scale_lr: false
  max_grad_norm: 1.0
  lr_scheduler: constant
  lr_warmup_steps: 0

model:
  act_fn: silu
  add_audio_layer: true
  attention_head_dim: 8
  block_out_channels: [320, 640, 1280, 1280]
  center_input_sample: false
  cross_attention_dim: 384
  down_block_types:
    [
      "CrossAttnDownBlock3D",
      "CrossAttnDownBlock3D",
      "CrossAttnDownBlock3D",
      "DownBlock3D",
    ]
  mid_block_type: UNetMidBlock3DCrossAttn
  up_block_types:
    [
      "UpBlock3D",
      "CrossAttnUpBlock3D",
      "CrossAttnUpBlock3D",
      "CrossAttnUpBlock3D",
    ]
  downsample_padding: 1
  flip_sin_to_cos: true
  freq_shift: 0
  in_channels: 13 # 49
  layers_per_block: 2
  mid_block_scale_factor: 1
  norm_eps: 1e-5
  norm_num_groups: 32
  out_channels: 4 # 16
  sample_size: 64
  resnet_time_scale_shift: default # Choose between [default, scale_shift]

  use_motion_module: true
  motion_module_resolutions: [1, 2, 4, 8]
  motion_module_mid_block: false
  motion_module_decoder_only: false
  motion_module_type: Vanilla
  motion_module_kwargs:
    num_attention_heads: 8
    num_transformer_block: 1
    attention_block_types:
      - Temporal_Self
      - Temporal_Self
    temporal_position_encoding: true
    temporal_position_encoding_max_len: 24
    temporal_attention_dim_div: 1
    zero_initialize: true
```

**Comment by chunyu-li** [comment]
The result:

https://github.com/user-attachments/assets/e7f57d34-6964-4006-becd-c0ad71f246a0

**Comment by Rupanshparth** [comment]
Yeah, I started getting good results now after setting the resume global step to 0 for the second stage, as mentioned earlier. The issue might have been with optimizer initialization when the global step was other than 0, the results I shared was when second stage training started with resume step of 360k
@chunyu-li
